cmake/GoogleCloudCppLibrary.cmake

```diff
@@ -57,6 +59,11 @@ function (google_cloud_cpp_add_ga_grpc_library library display_name)
     endif ()
     set(DOXYGEN_EXCLUDE_SYMBOLS "internal")
     set(DOXYGEN_EXAMPLE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/quickstart")
+    unset(GOOGLE_CLOUD_CPP_DOXYGEN_EXTRA_INCLUDES)
```

Comment:
Starting with CMake 3.12 we can use `list(TRANSFORM ...)`:

https://cmake.org/cmake/help/latest/command/list.html#transform

```CMake
set(GOOGLE_CLOUD_CPP_DOXYGEN_EXTRA_INCLUDES "${opt_CROSS_LIB_DEPS}")
list(TRANSFORM
      GOOGLE_CLOUD_CPP_DOXYGEN_EXTRA_INCLUDES
      PREPEND "${PROJECT_BINARY_DIR}/google/cloud/")
```